rust-runtime/aws-smithy-wasm/src/wasi_adapter.rs

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+//! HTTP WASI Adapter
+
+use aws_smithy_client::{erase::DynConnector, http_connector::HttpConnector};
+use aws_smithy_http::{body::SdkBody, result::ConnectorError};
+use bytes::Bytes;
+use http::{Request, Response};
+use std::{
+    sync::Arc,
+    task::{Context, Poll},
+};
+use tower::Service;
+use wasi_preview2_prototype::http_client::DefaultClient;
+
+pub fn wasi_connector() -> HttpConnector {
+    HttpConnector::ConnectorFn(Arc::new(|_, _| Some(DynConnector::new(Adapter::default()))))
+}
+
+#[derive(Default, Debug, Clone)]
+/// HTTP Service Adapter used in WASI environment
+pub struct Adapter {}
+
+impl Service<Request<SdkBody>> for Adapter {
+    type Response = Response<SdkBody>;
+
+    type Error = ConnectorError;
+
+    #[allow(clippy::type_complexity)]
+    type Future = std::pin::Pin<
+        Box<dyn std::future::Future<Output = Result<Self::Response, Self::Error>> + Send + 'static>,
+    >;
+
+    fn poll_ready(&mut self, _cx: &mut Context<'_>) -> Poll<Result<(), Self::Error>> {
+        Poll::Ready(Ok(()))
+    }
+
+    fn call(&mut self, req: http::Request<SdkBody>) -> Self::Future {
+        println!("Adapter: sending request...");
+        let client = DefaultClient::new(None);
```

Comment:
do we want to be sharing the client between requests? (I don't know anything about WASI here)

Reply:
there is no initialization step at this moment (https://github.com/WebAssembly/wasi-http/blob/main/wit/outgoing-handler.wit), so probably it is not worth to deal with that complexity now. besides, the webassembly host (wasmtime in our tests) will be responsible to make any optimization there.